spec/services/newsletter_events_job_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+describe ::NewsletterEventsJob do
+  it 'should create a campaign'
+
+  it 'should collect all events' do
+    FactoryBot.create_list(:organisation, 2)
+    FactoryBot.create(:event, organisation: Organisation.first)
+    FactoryBot.create(
+        :event,
+        start_date: Date.today - 4,
+        organisation: Organisation.last
+    )
+    events = Event.where('start_date >= ?', Date.today).reduce([]) do |data, event|
+      data << NewsletterEventsJob.format_event(event)
+    end
+    expect(events).to eq(NewsletterEventsJob.events)
+    expect(NewsletterEventsJob.events.count).to eq(1)
+    expect(Event.count).to eq(2)
```

Comment:
interesting approach - it seems like you're adding class methods to the job so that components of it can be tested.

I'd be tempted to minimize the job interface to just one method, and then have a test for the output, e.g. that all the events are appearing in the post to mailchimp - or even in the call to gibbon - more of an outside in approach if you know what I mean ...?

Reply:
You got me here, yup it was easy for me to test it in that way... yes I get it, but to have only one method is it beneficial for us in the long term, just asking because I don't know about patterns regarding this decision :+1: